File: src/006/p006.hs
sumOfSquares :: Integral n => [n] -> n
sumOfSquares ns = sum $ map (^2) ns

squareOfSum :: Integral n => [n] -> n
squareOfSum ns = (sum ns)^2

difference :: Integral n => [n] -> n
difference ns = squareOfSum ns - sumOfSquares ns

main :: IO ()
main = print $ difference [1..100]

File: src/006/p006.zig
const std = @import("std");
const warn = std.debug.warn;

pub fn main() void {
    var sum_of_squares: u64 = 0;
    var sum: u64 = 0;
    var j: u64 = 1;
    var res: u64 = while (j <= 100) : (j += 1) {
        sum_of_squares += j * j;
        sum += j;
    } else (sum * sum) - sum_of_squares;
    warn("{}\n", res);
}
